### DISCUSSION TITLE
Problem with position using LPS

### DISCUSSION THREAD

**Discussion by AngelS18** [bitcraze/discussions#732] [Q&A]
Hi, I am trying to make a flight with 2 drones using for closed loop control the Loco positioning system in TDoA 2 mode. When I tested it with a single drone in TWR mode, it worked very well thanks to the implemented Kalman filter, taken from:
https://github.com/bitcraze/crazyflie-lib-python/blob/master/examples/positioning/initial_position.py

I would like to know if there are any developments to improve the positioning for TDoA 2 mode or where I could take the information from to try to improve the position control.

I understand that in TDoA 2 mode it only listens passively and I guess that will have to affect, I would like to understand how I could improve the filter or with what I can make a better position control.

**Comment by knmcguire** [comment]
Your title implies that you are having a problem. Do you have an issue with loco positioning and the crazyflie specifically or is this just a question.

There is TDOA3 that is an improvement of TDOA2. You can find the technical information about it here. https://www.bitcraze.io/documentation/repository/lps-node-firmware/master/

If you have still questions remaining after reading the doc, please let us know here.

**Reply by AngelS18** [reply]
ok, thank you for your answer.
